daml-lf/transaction/src/main/scala/com/digitalasset/daml/lf/transaction/Node.scala

```diff
@@ -81,6 +81,11 @@ object Node {
       chosenValue: Val,
       stakeholders: Set[Party],
       signatories: Set[Party],
+      /** Note that here we have both actors and controllers because we use this
+        * data structure _before_ we validate the transaction. However for every
+        * valid transaction the actors must be the same as the controllers. This
+        * is why we removed the controllers field in transaction version 6.
+        */
       controllers: Set[Party],
```

Comment:
What if we delete this field from the case class?

Reply:
we can't if we keep the validation separate.

Reply:
Maybe we should collapse the data cases here, then: make this `controllersDifferFromActors: Boolean`, add `def controllers: actingParties.type` to the class body, and reject if the boolean is true.

Reply:
@S11001001 I'm not sure I understand the proposal. Could you clarify with a code suggestion, maybe?

Reply:
and here

```suggestion
       controllersDifferFromActors: Boolean,
```